src/XrdHttp/XrdHttpReq.cc

```diff
@@ -1180,43 +1201,28 @@ int XrdHttpReq::ProcessHTTPReq() {
             xrdreq.dirlist.dlen = htonl(l);
 
             if (!prot->Bridge->Run((char *) &xrdreq, (char *) res.c_str(), l)) {
-              prot->SendSimpleResp(404, NULL, NULL, (char *) "Could not run request.", 0, false);
-              return -1;
+              XrdHttpReq::mapXrdErrorToHttpStatus();
```

Comment:
You can use just `mapXrdErrorToHttpStatus();` here, as we are inside `XrdHttpReq` anyway.

Reply:
Addressed in the latest commit (plus other, similar locations).